File: ccif/src/java/other/org/himalay/ccif/ByteArrayWrapper_ui8.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class ByteArrayWrapper_ui8 extends BinStruct implements PublicBinMsg {




    // members variables
    // data
    public ByteArray data ;

    public ByteArrayWrapper_ui8 () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize data
        data= new ByteArray();
        data.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read data
        {
            retVal+=data.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for data**/
        }

        // write data
        {
            retVal += data.write(ostream);
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("ByteArrayWrapper_ui8\n");
        dc.increaseIndent();
        int retVal= 0;
        // write data
        dc.indent();
        dc.getPs().print("data: "+data.getSize()+"(0x"+Integer.toHexString(data.getSize())+")\n");
        this.data.dump(dc);
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for data
    //public ByteArray getData()
    //{
    //    return data ;
    //}


    // Setter for data
    //public void setData(ByteArray val)
    //{
    //    this.data= val;
    //}


    public void setData(byte[] val)
    {
        this.data.setData(val);
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/ByteArrayWrapper_ui8.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ByteArrayWrapper_ui8.h"

#ifdef org_himalay_ccif_ByteArrayWrapper_ui8__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ByteArrayWrapper_ui8::ByteArrayWrapper_ui8() {
    references= 0;


    // data
    org_himalay_msgs_runtime_ByteArray_NEW(data);
    data->setSizeType("FIRST_UI8");
}
// Destructor
ByteArrayWrapper_ui8::~ByteArrayWrapper_ui8() {
    // data
    org_himalay_msgs_runtime_ByteArray_DELETE( data);
}

int ByteArrayWrapper_ui8::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // data
    {
        retVal+=data->read(istream);
    }
    return retVal;
}

int ByteArrayWrapper_ui8::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read data
    {
        retVal+=data->read(istream);
    }

    return retVal;
}

int ByteArrayWrapper_ui8::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for data**/
    }

    int retVal= 0;
    // write data
    {
        retVal += data->write(ostream);
    }

    return retVal;
}

int ByteArrayWrapper_ui8::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ByteArrayWrapper_ui8\n");
    dc.increaseIndent();
    int retVal= 0;
    // write data
    dc.indent();
    dc.getPs().print("data");
    data->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for data
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE ByteArrayWrapper_ui8::getData()
//{
//return this->data;
//}

// Setter for data
//void ByteArrayWrapper_ui8::setData(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->data= val;
//}

int ByteArrayWrapper_ui8::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_ByteArrayWrapper_ui8__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ByteArrayWrapper_ui8* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ByteArrayWrapper_ui8* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code